api/src/main/java/jakarta/data/repository/Query.java

```diff
@@ -20,49 +20,78 @@
- * JPQL queries if backed by a Jakarta Persistence provider,
- * and otherwise must support SQL queries.</p>
- *
- * <h2>Parameters</h2>
- *
- * <p>The query language can used named parameters or positional parameters.</p>
- *
- * <p><b>Named parameters</b> are referred to by name within the query language.
- * The {@link Param} annotation annotates method parameters to bind them to a named parameter name.
- * The {@code Param} annotation is unnecessary for named parameters when the method parameter name
- * matches the query language named parameter name and the application is compiled with the
- * {@code -parameters} compiler option that makes parameter names available
- * at run time. When the {@code Param} annotation is not used, the Jakarta Data provider must
- * interpret the query by scanning for the delimiter that is used for positional parameters.
- * If the delimiter appears for another purpose in a query that requires named parameters,
- * it might be necessary for the application to explicitly define the {@code Param} in order to
- * disambiguate.</p>
- *
- * <p><b>Positional parameters</b> are referred to by a number that corresponds to the
- * numerical position, starting with 1, of the repository method parameters.</p>
+ * <p>Annotates a repository method as a query method, specifying a query written in Jakarta Data Query Language (JDQL)
+ * or in Jakarta Persistence Query Language (JPQL). A Jakarta Data provider is not required to support the complete JPQL
+ * language, which targets relational data stores.</p>
+ *
+ * <p>The required {@link #value} member specifies the JDQL or JPQL query as a string. When {@linkplain Page pagination}
+ * is used, the {@link #count} member optionally specifies a query which returns the total number of elements satisfying
+ * the query.</p>
+ *
+ * <p>For {@code select} statements, the return type of the query method must be consistent with the type returned by
+ * the query. For {@code update} or {@code delete} statements, it must be {@code void}, {@code int} or {@code long}.</p>
+ *
+ * <p>Compared to SQL, JDQL allows an abbreviated syntax for {@code select} statements:</p>
+ * <ul>
+ * <li>The {@code from} clause is optional in JDQL. When it is missing, the queried entity is determined by the return
+ *     type of the repository method, or, if the return type is not an entity type, by the primary entity type of the
+ *     repository.</li>
+ * <li>The {@code select} clause is optional in both JDQL and JPQL. When it is missing, the query returns the queried
+ *     entity.</li>
+ * </ul>
+ *
+ * <p>A query might involve:</p>
+ * <ul>
+ * <li>named parameters of form {@code :name} where the labels {@code name} are legal Java identifiers, or </li>
+ * <li>ordinal parameters of form {@code ?n} where the labels {@code n} are sequential positive integers starting
+ *     from {@code 1}.</li>
+ * </ul>
+ * <p>A given query may not mix named and ordinal parameters.</p>
```

Comment:
This means the same thing? I mean, I don't care but....

Reply:
> This means the same thing? I mean, I don't care but....

Unfortunately there are a lot of people who read/use the word "may" as meaning "might" in common language so we are told to generally avoid it in specifications and use "must" to avoid confusion.

If someone does read it as "might" in this case, they will consider the statement to be informational rather than a requirement.